utils/kubernetes/volumes_test.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2023 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubernetes
+
+import (
+	"sort"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+)
+
+func TestReplaceOrAddVolume(t *testing.T) {
+	podSpec := v1.PodSpec{Volumes: []v1.Volume{
+		{Name: "volume1", VolumeSource: v1.VolumeSource{ConfigMap: &v1.ConfigMapVolumeSource{
+			LocalObjectReference: v1.LocalObjectReference{Name: "cm1"},
+		}}},
+		{Name: "volume2", VolumeSource: v1.VolumeSource{ConfigMap: &v1.ConfigMapVolumeSource{
+			LocalObjectReference: v1.LocalObjectReference{Name: "cm2"},
+		}}},
+	}}
+	volumes := []v1.Volume{
+		{Name: "volume1", VolumeSource: v1.VolumeSource{ConfigMap: &v1.ConfigMapVolumeSource{
+			LocalObjectReference: v1.LocalObjectReference{Name: "cmA"},
+		}}},
+		{Name: "volume2", VolumeSource: v1.VolumeSource{ConfigMap: &v1.ConfigMapVolumeSource{
+			LocalObjectReference: v1.LocalObjectReference{Name: "cmB"},
+		}}},
+		{Name: "volume3", VolumeSource: v1.VolumeSource{ConfigMap: &v1.ConfigMapVolumeSource{
+			LocalObjectReference: v1.LocalObjectReference{Name: "cmC"},
+		}}},
+	}
+
+	AddOrReplaceVolume(&podSpec, volumes...)
+
+	assert.Len(t, podSpec.Volumes, 3)
+	sort.Slice(podSpec.Volumes, func(i, j int) bool {
+		return podSpec.Volumes[i].Name < podSpec.Volumes[j].Name
+	})
+	assert.Equal(t, "cmA", podSpec.Volumes[0].ConfigMap.Name)
+	assert.Equal(t, "cmB", podSpec.Volumes[1].ConfigMap.Name)
+	assert.Equal(t, "cmC", podSpec.Volumes[2].ConfigMap.Name)
```

Comment:
if my understanding is correct, the volumes provided in the podSpec will be removed?

Reply:
no! we merge user's volumes with the operator's

Reply:
shouldn't we have 5 volume in this test case then?

Reply:
No, volume1 and volume2 are in both containers.